### ISSUE TITLE
NMTran multi-record support: `$PROB`, `$EST`, `$SIM`.

### ISSUE THREAD

**Issue by king-of-poppk** [pharmpy/pharmpy#2061]
### Feature request

What is the current status of multi-record support of the `NMTranControlStream` structure? Would you be interested in a more generic implementation? Pharmpy's high-level would be unchanged of course.

**Comment by rikardn** [comment]
Do you mean support for multiple $PROBLEM? Do you have a use case in mind?

**Comment by rikardn** [comment]
And sure that sounds interesting. Let me know if you want to setup a meeting to discuss.

**Comment by vrognas** [comment]
This would be valuable, especially with support for MSF-files

**Comment by vrognas** [comment]
Check out my PR #2077, it has some initial basic logic to handle multiple $PROBLEM records.

**Comment by rikardn** [comment]
@king-of-poppk  Yes, `_active_problem` was a very old idea that is not in use and we will not be using. @vrognas has added an option to state the index of the `$PROBLEM` for some methods. A better long term solution could be to implement indexing and slicing of records. Then we could for example have a method to give the stream of all records in a certain `$PROBLEM`. Perhaps indexing could be both `cs['ESTIMATION']` and `ci[(0, 'ESTIMATION')]`  for all estimation records in the first `$PROBLEM`. There might be other ideas.

**Comment by king-of-poppk** [comment]
Slicing the control stream would be interesting. Not sure though how one would keep track of preceding and succeeding comments? Indexing could also be done `cs.problem(i).get_records(...)` or similar, just to put another idea in the basket.
